File: 프로그래머스/2/12978. 배달/배달.js
function solution(N, road, K) {
    road.sort((a,b)=>b[2]-a[2])
    const arr = Array.from({length:N},(fv,fi)=>Array.from({length:N},(v,i)=>fi===i?-1:0))
    const vArr = Array.from({length:N},(_,i)=>(i===0?0:Infinity))
    const visited = {}
    
    for(const r of road){
        arr[r[0]-1][r[1]-1] = r[2];
        arr[r[1]-1][r[0]-1] = r[2];
    }
    
    const queue = [0];
    while(queue.length){
        const cur = queue.shift();
        visited[cur] = true;
        for(let i=0; i<N; i+=1){
            if(arr[cur][i]>0){
                const newValue = vArr[cur]+arr[cur][i]
                
                if(vArr[i]>newValue){
                    queue.push(i);
                    vArr[i] = newValue
                }
            }
        }
    }
    
    return vArr.filter(v=>v<=K).length;
}

File: 프로그래머스/2/12978. 배달/배달.py
from collections import defaultdict
import heapq

def solution(N, road, K):
    nodes = defaultdict(list)
    for n1, n2, weight in road:
        nodes[n1].append((n2, weight))
        nodes[n2].append((n1, weight))
    visited = [False]*(N+1)
    costs = [float('inf')]*(N+1)
    costs[1] = 0
    queue = [(0,1)]
    
    while queue:
        cur_weight, cur_node = heapq.heappop(queue)
        if visited[cur_node]: continue
        visited[cur_node] = True
        for next_node, next_cost in nodes[cur_node]:
            new_cost = cur_weight + next_cost
            if new_cost < costs[next_node]:
                costs[next_node] = new_cost
                heapq.heappush(queue, (new_cost, next_node))
    
    result = 0
    for i in costs:
        if i<=K: result+=1
    return result